Pull Request for Electron Microscopy Particle Segmentation #234

achrekarom12 opened the pull request:
# Pull Request for DL-Simplified 💡


## Issue Title :  Electron Microscopy Particle Segmentation

- **Info about the related issue (Aim of the project)** : Create a DL model, which will segment different particles.
- **Name:** Om Achrekar
- **GitHub ID:** achrekarom12
- **Email ID:** achrekarom@gmail.com
- **Idenitfy yourself: (Mention in which program you are contributing in. Eg. For a JWOC 2022 participant it's, `JWOC Participant`)** SSoC '23


<!-- Mention the following details and these are mandatory -->

Closes: #234 

### Describe the add-ons or changes you've made 📃
ipynb file for the mentioned issue title which contains segmentation model with the help of DeepLabv3+.

## Type of change ☑️

What sort of change have you made:
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested? ⚙️

Describe how it has been tested
Describe how have you verified the changes made

## Checklist: ☑️
- [x] My code follows the guidelines of this project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly wherever it was hard to understand.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added things that prove my fix is effective or that my feature works.
- [x] Any dependent changes have been merged and published in downstream modules.

abhisheks008 commented on Electron Microscope Particle Segmentation/Model/Electron Microscopy Particle Segmentation.ipynb:
1. Try to implement few more Data visualization based on the dataset. Your project looks 70% completed.
2. Can you use 2 more deep learning techniques on this project, to get a better comparison between the models based on the performance metrices.
3. Include all the visualization images inside the Images folder and also put them in the README.md file under the **VISUALIZATION** section.

abhisheks008 commented on Electron Microscope Particle Segmentation/requirements.txt:
Add the versions of the libraries.

abhisheks008 reviewed (changes requested):
@achrekarom12

achrekarom12 commented:
Hey @abhisheks008, I have committed the changes to the repository, should I have to create a new PR for it?
And regarding the changes I have made:
1. I have implemented one additional model which is UNet. I feel the dataset was incomplete (correct me if I'm wrong) so I have to annotate the masks manually. It took time. There are few more DL techniques such as PSPNet, ColonSegNet, FCN and Mask R-CNN but there were different requirements for them.
2. Rest of the things I have done as instructed by you. Added more viz., updated README.md and Requirements.txt with necessary changes.

Let me know if any further changes you would want in this.
Thankyou!

abhisheks008 reviewed (approved):
It looks better than the previous version. Looks good to me.
Your PR is approved and ready to be merged.

@achrekarom12
